### ISSUE TITLE
Missing GROUP_BY support in Multisearch requests

### ISSUE THREAD

**Issue by jarto666** [DAXGRID/typesense-dotnet#247]
Can support for group_by parameter in multisearch requests be implemented?

**Comment by Spaggel** [comment]
I was also trying to get grouped results. You can do something like this:
```csharp
var query = new GroupedSearchParameters("search term", "query_field", "GROUP_BY_FIELD"); # GROUP_BY_FIELD must be a facet field
var searchResult = await client.SearchGrouped<Product>("collection_name", query);
foreach (var result in searchResult.GroupedHits)
{
    foreach (var resultHit in result.Hits)
    {
        Console.WriteLine(resultHit.Document.Id + ": " + resultHit.Document.Title);   
    }
}
```

**Comment by jarto666** [comment]
Just for clarification: ideally it should be flexible to support making both - regular and grouped requests in the same scope of multisearch.

Thanks!

**Comment by jarto666** [comment]
@Spaggel I need not only grouped results, but also make a couple of other ones in the scope of a single multisearch request. 

See: https://typesense.org/docs/27.0/api/federated-multi-search.html

**Comment by Spaggel** [comment]
@jarto666 is this what you're looking for? https://github.com/DAXGRID/typesense-dotnet#multi-search-documents

**Comment by jarto666** [comment]
@Spaggel Yes, but current MultiSearchParameters implementation does not support GroupBy

**Comment by jarto666** [comment]
@runeanielsen I wanted to create a PR for you to review and give your thoughts on the implementation, but seems like I need specific permissions for this.

So, we could add those to MultiSearchParameters or try to inherit those 3 from GroupedSearchParameters (notice group_missing_values which is also missing from current codebase). Though, inheritance would also require revision of GroupedSearchParameters constructors, since groupBy in this case would become optional. 
`
[JsonPropertyName("group_by")]
public string GroupBy { get; set; }

[JsonPropertyName("group_limit")]
public int? GroupLimit { get; set; }

[JsonPropertyName("group_missing_values")]
public bool? GroupMissingValues { get; set; }
`

And this would allow us to simply extend MultiSearchResult with GroupedHits:

`
public record MultiSearchResult<T>
{
    [JsonPropertyName("facet_counts")]
    public IReadOnlyList<FacetCount>? FacetCounts { get; init; }

    [JsonPropertyName("found")]
    public int? Found { get; init; }

    [JsonPropertyName("hits")]
    public IReadOnlyList<Hit<T>>? Hits { get; init; }
    
    [JsonPropertyName("grouped_hits")]
    public IReadOnlyList<GroupedHit<T>> GroupedHits { get; init; }
`

Not the cleanest solution architecturally, the main issue with those parameters duplication, but up for you to consider it. Tested and it's working.

Thanks!

**Comment by runeanielsen** [comment]
@jarto666 I've released a new version (7.19.0). Thank you for you contribution. :)

// Rune
